lib/CompilerImpl/JITCompiler.cpp

```diff
@@ -243,314 +242,547 @@ JITCompiler::runOptimizer(const std::filesystem::path &src_IR,
-    return failureFileName;
-  }
 
-  // In run twice mode, we want to make sure the output is bit-by-bit
-  // equivalent if we run the pass manager again, so setup two buffers and
-  // a stream to write to them.
-  llvm::SmallVector<char, 0> Buffer;
-  llvm::SmallVector<char, 0> CompileTwiceBuffer;
-  std::unique_ptr<llvm::raw_svector_ostream> BOS;
-  llvm::raw_ostream *OS = nullptr;
-
-  // Write bitcode or assembly to the output as the last step...
-  OS = &Out->os();
-  if (RunTwice) {
-    BOS = std::make_unique<llvm::raw_svector_ostream>(Buffer);
-    OS = BOS.get();
-  }
-  if (OutputAssembly) {
-    if (EmitSummaryIndex) {
-      report_error("Text output is incompatible with -module-summary");
+    // In run twice mode, we want to make sure the output is bit-by-bit
+    // equivalent if we run the pass manager again, so setup two buffers and
+    // a stream to write to them.
+    llvm::SmallVector<char, 0> Buffer;
+    llvm::SmallVector<char, 0> CompileTwiceBuffer;
+    std::unique_ptr<llvm::raw_svector_ostream> BOS;
+    llvm::raw_ostream *OS = nullptr;
+
+    // Write bitcode or assembly to the output as the last step...
+    OS = &Out->os();
+    if (RunTwice) {
+      BOS = std::make_unique<llvm::raw_svector_ostream>(Buffer);
+      OS = BOS.get();
     }
-    if (EmitModuleHash) {
-      report_error("Text output is incompatible with -module-hash");
+    if (OutputAssembly) {
+      if (EmitSummaryIndex) {
+        report_error("Text output is incompatible with -module-summary");
+      }
+      if (EmitModuleHash) {
+        report_error("Text output is incompatible with -module-hash");
+      }
+      Passes.add(createPrintModulePass(*OS, "", PreserveAssemblyUseListOrder));
     }
-    Passes.add(createPrintModulePass(*OS, "", PreserveAssemblyUseListOrder));
-  }
 #if LLVM_VERSION_MAJOR < 16
-  else if (OutputThinLTOBC) {
-    Passes.add(createWriteThinLTOBitcodePass(*OS));
-  }
-#endif
-  else {
-    Passes.add(createBitcodeWriterPass(*OS, PreserveBitcodeUseListOrder,
-                                       EmitSummaryIndex, EmitModuleHash));
-  }
-
-#ifdef VC_DEBUG
-  // Before executing passes, print the final values of the LLVM options.
-  llvm::cl::PrintOptionValues();
+    else if (OutputThinLTOBC) {
+      Passes.add(createWriteThinLTOBitcodePass(*OS));
+    }
 #endif
+    else {
+      Passes.add(createBitcodeWriterPass(*OS, PreserveBitcodeUseListOrder,
+                                         EmitSummaryIndex, EmitModuleHash));
+    }
 
-  // If requested, run all passes again with the same pass manager to catch
-  // bugs caused by persistent state in the passes
-  if (RunTwice) {
-    std::unique_ptr<llvm::Module> module2(llvm::CloneModule(*module));
-    Passes.run(*module2);
-    CompileTwiceBuffer = Buffer;
-    Buffer.clear();
-  }
+    // #ifdef VC_DEBUG
+    //  Before executing passes, print the final values of the LLVM options.
+    llvm::cl::PrintOptionValues();
+    // #endif
```

Comment:
why is this no longer only enabled in debug mode?

Reply:
No clue honestly

Reply:
Can we restore the debug guards?